measurements/api/private.py

```diff
@@ -44,34 +44,49 @@ def api_private_stats_by_month(orm_stat):
     end_date = datetime(now.year, now.month, 1)
     start_date = end_date - relativedelta(months=24)
 
-    r = current_app.db_session.query(
-            func.date_trunc('month', Report.test_start_time).label('test_start_month'),
-            orm_stat
-        ).filter(Report.test_start_time >= start_date).filter(Report.test_start_time < end_date
-        ).group_by('test_start_month')
+    s = select([
+        func.date_trunc('month', sql.text('bucket_date')).label('bucket_month'),
+        sql.text(orm_stat)
+    ]).where(and_(
+        sql.text("bucket_date > :start_date"),
+        sql.text("bucket_date < :end_date")
+    )).group_by(
+        'bucket_month'
+    ).select_from(sql.table('ooexpl_bucket_msm_count'))
+
+    r = current_app.db_session.execute(s, {'start_date': start_date, 'end_date': end_date})
     result = [{
         'date': (bkt + relativedelta(months=+1, days=-1)).strftime("%Y-%m-%d"),
         'value': value,
     } for bkt, value in sorted(r)]
-    return jsonify(result)
+    return result
 
 @api_private_blueprint.route('/asn_by_month')
 def api_private_asn_by_month():
-    # The query takes ~6s on local SSD @ AMS on 2018-04-04.
-    # It was taking ~45s when it was fetching all the table from DB and doing grouping locally.
-    return api_private_stats_by_month( func.count(distinct(Report.probe_asn)) )
+    return jsonify(api_private_stats_by_month('COUNT(DISTINCT probe_asn)'))
 
 @api_private_blueprint.route('/countries_by_month')
 def api_private_countries_by_month():
-    # The query takes ~10s on local SSD @ AMS on 2018-04-04.
-    # It was taking ~25s when it was fetching all the table from DB and doing grouping locally.
-    return api_private_stats_by_month( func.count(distinct(Report.probe_cc)) )
+    return jsonify(api_private_stats_by_month('COUNT(DISTINCT probe_cc)'))
 
 @api_private_blueprint.route('/runs_by_month')
 def api_private_runs_by_month():
     # The query takes ~6s on local SSD @ AMS on 2018-04-04.
     # It was taking ~20s when it was fetching all the table from DB and doing grouping locally.
-    return api_private_stats_by_month( func.count() )
+    now = datetime.now()
+    end_date = datetime(now.year, now.month, 1)
+    start_date = end_date - relativedelta(months=24)
+
+    r = current_app.db_session.query(
+            func.date_trunc('month', Report.test_start_time).label('test_start_month'),
+            func.count()
+        ).filter(Report.test_start_time >= start_date).filter(Report.test_start_time < end_date
+        ).group_by('test_start_month')
+    result = [{
+        'date': (bkt + relativedelta(months=+1, days=-1)).strftime("%Y-%m-%d"),
+        'value': value,
+    } for bkt, value in sorted(r)]
```

Comment:
I'm failing to understand the rationale of this change, can you explain?

Reply:
This is basically just restoring the code for this particular API endpoint to how it was before.

See: https://github.com/ooni/api/pull/104#issuecomment-527965586 & https://github.com/ooni/api/pull/104#issuecomment-528823688 for a more in-depth explaination as to why we want to preseve the old slow query for this endpoint vs using the materialized table for the others (tl;dr the materialized tables don't have enough information in them to be the same result).